src/meta/src/barrier/mod.rs

```diff
@@ -600,13 +600,7 @@ where
             let info = self
                 .resolve_actor_info(&mut checkpoint_control, &command)
                 .await;
-            // When there's no actors exist in the cluster, we don't need to send the barrier. This
-            // is an advance optimization. Besides if another barrier comes immediately,
-            // it may send a same epoch and fail the epoch check.
-            if info.nothing_to_do() {
```

Comment:
Since this is already filtered in `inject_barrier` and `collect_barrier`, this optimization is no longer needed. After that we can advance epochs in order based on the barrier complete channel, simply advance epochs here will break the epoch updating orders. This is the reason why the previous ci failed: https://buildkite.com/risingwavelabs/pull-request/builds/15220#0185864f-496d-461b-98aa-f23dc01191a2

Reply:
+1